### ISSUE TITLE
Automated building and deployment of documentation

### ISSUE THREAD

**Issue by mikadou** [RigsOfRods/rigs-of-rods#623]
As discussed in https://github.com/RigsOfRods/rigs-of-rods/pull/576 it would be nice to have documentation to be build automatically for each commit and deployed to Github pages.

In order to achieve this, it will be necessary to grant commit rights to Travis build service.

Furthermore there is a decision to be made where to place the generated docs since Github supports 2 different kinds of pages:

_User/organizational pages_
These are placed under rigsofrods.github.io and have their own repository. The good thing is that Travis would only need to be granted access to this single repository. The downside is that each update of the source code will also cause a commit to the project pages repository (i.e. there will be a lot of "[Travis] Autoupdated docs" commit messages in https://github.com/RigsOfRods/rigsofrods.github.io.

_Project pages_
These are placed under rigsofrods.github.io/rigs-of-rods, i.e. there can be a project page for each repository (rigs-of-rods in this case). Project pages do not live in a separate repository, but are simply placed inside the corresponding project repo on a special branch named gh-pages. However as far as I can see it is not possible to restrict commit access to single branches, therefore Travis would need access for the entire rigs-of-rods repo.

I favour the _project pages_ solution as it doesn't clutter the commit log of the rigsofrods.github.io repository. What's your opinion on this?

**edit**
A third option would be to create a new repository specifically for documentation. The project pages (i.e. gh-pages branch) of this new repository would be used to host the documentation files. The master branch of this repo will probably stay empty.

**Comment by mikadou** [comment]
I decided to go with the third option and created a pull request. Currently the auto-built documentation is pushed to [this repository](https://github.com/mikadou/ror-documentation) under my account. It also contains a description how things were setup.
The documentation is available at http://mikadou.github.io/ror-documentation.

I suggest we see how this works and if it does well, we can move the `ror-documentation` repository under the official RigsOfRods organization account.

**Comment by mikadou** [comment]
Apparently I was too optimistic, the PR I merged does not automatically deploy the built documentation. The problem is that the encrypted environemt variables were setup for my own forked repository, but not the official rigs-of-rods repository.
I created a new branch https://github.com/RigsOfRods/rigs-of-rods/tree/deploy-docs because I need the travis builds to be executed inside the official repository, but I don't want to clutter the master branch while tinkering.
